src/univariate/continuous/uniform.jl

```diff
@@ -101,6 +101,36 @@ function mgf(d::Uniform, t::Real)
     v = (a + b) * t / 2
     exp(v) * (sinh(u) / u)
 end
+function cgf_uniform_around_zero_kernel(x)
+    # taylor series of (exp(x) - x - 1) / x
+    T = typeof(x)
+    a0 = inv(T(2))
+    a1 = inv(T(6))
+    a2 = inv(T(24))
+    a3 = inv(T(120))
+    x*@evalpoly(x, a0, a1, a2, a3)
```

Comment:
I changed a bit what to put in the kernel and I believe this is sufficiently accurate for any floating point number type in the region where we apply it. 
The region of applicability is `abs(x) < sqrt(eps(T))` and I will assume `x > 0` for readability. The error from truncating the Taylor series is `err(x) ~ O(x^5)`, so we find:
```
err(x) 
~ O(x^5)
~ O(x^2) * O(x) * O(x^2) 
< O(x^2) * O(x) * sqrt(eps(T))^2 # region assumption
~ O(x^2) * O(eps(x))                   # eps(x)  ≈ x * eps(T)`
```
Since `cgf_uniform_around_zero_kernel(x) = x + O(x^2)` this error is way below machine precision `eps(x + O(x^2))`. This argument is not specific to `Float64`, it only assumes `eps(x)  ≈ x * eps(T)` which should be true for any floating point type I think. See also tests against `BigFloat` below.
First and second order derivatives are also fine by similar argument. 
Of course, higher order derivatives will at some point become zero since this is a finite degree polynomial.

So I would like to leave the kernel as is. If people want to optimize it further or better support higher order AD that can be a future PR, what do you think @devmotion ?

Reply:
I'm still wondering if there's another approach that would avoid that we have to work with the approximations and branching in Distributions directly (instead of reusing existing functionality in LogExpFunctions, SpecialFunctions etc.). But I think it's the best we can do for now at least.

Reply:
I agree, it would be nice if we could pull this function in from somewhere else. I did look through LogExpFunctions before coding this, but could not see a way to use LogExpFunctions here. For SpecialFunctions I don't have high hopes either, since our function is not really special, but expressible in terms of `exp` and polynomials.